### ISSUE TITLE
Parquet bool column SortOrder being UNSIGNED?

### ISSUE THREAD

**Issue by jp0317** [apache/arrow-rs#6544]
**Describe the bug**
Hi, please close if it's intended. It seems that SortOrder for bool is [UNSIGNED](https://github.com/jp0317/arrow-rs/blob/master/parquet/src/basic.rs#L625) in rust but [SIGNED](https://github.com/apache/arrow/blob/main/cpp/src/parquet/types.cc#L272-L277) in c++ implementation. Should the rust implementation also be SIGNED?

**To Reproduce**
n/a

**Expected behavior**
Consistency?

**Additional context**
n/a

**Comment by tustvold** [comment]
Given a boolean doesn't have a sign, I would have expected it to be unsigned

**Comment by jp0317** [comment]
thanks @tustvold for the reply. Yeah that makes sense to me too, but both c++ and java implementation go with `SIGNED`, which can [affect if the column statistics are considered correct](https://github.com/apache/arrow/blob/main/cpp/src/parquet/metadata.cc#L1489).

**Comment by tustvold** [comment]
That is because older versions of those libraries used to write invalid statistics, see a few lines earlier where it checks the parquet version - https://github.com/apache/arrow/blob/main/cpp/src/parquet/metadata.cc#L1482

This codepath should not be being encountered with data written by parquet-rs AFAICT

**Comment by jp0317** [comment]
yeah, i got that. But wouldn't it be better if parquet-rs can read files written by other implementations?

**Comment by tustvold** [comment]
We faithfully reproduce the statistics as written within the file, one could raise the prospect of adding similar validity checking, perhaps you might file a separate issue for this if there isn't one already. However, as for the topic of this issue, perhaps you could provider a reproducer of the actual issue you are running into? AFAICT we are writing unsigned, which the type objectively is, and some other implementations choose to write signed, likely for backwards compatibility, but I can't see what issue this could cause.

**Comment by alamb** [comment]
Maybe we can try to create a bug reproducer using a system like https://datafusion.apache.org/ or something else that uses the rust parquet reader (i.e. see if whatever statistics case we are concerned about can generate incorrect results)
